### PR TITLE
Feature/53 integrating database into tracex

### LINKED ISSUE #53: Integrating Database into TracEX
As a developer, I want to have access to an integrated database in the TracEX tool, to simplify file handling and lay the foundation for future features that make using the TracEX more comfortable for the user.

**Description:** Using concise bundles of data in form of database objects that follow a defined model is much easier than managing files in directories. Especially since Django supports Database integration natively. Reading and following the detailed Django Documentation is the way to go here. Things that should be stored in the DB (this list is expandable): Prompts, User Input (Patient Journeys, Configuration), extracted traces, combined event logs, DFG images.

**Estimated Size:** L

# Acceptance Criteria
- [x] Database is set up and connected to project
- [x] all necessary models are defined
- [x] user input is stored in DB
- [ ] prompts are stored and loaded from DB
- [ ] Extracted Traces are stored and loaded from DB
- [ ] combined event logs are stored and loaded from DB
- [ ] DFG Images are stored and loaded from DB

### PR DESCRIPTION
**Pull request by FR-SON**
This PR adds the database integration for TracEX. Uploaded patient journeys and all extraction results are saved. 
After seeing the extraction results, users can decide whether to save or discard them. 
There are some issues remaining (listed below). Feel free to comment on them, suggest solutions, implement them yourself or create issues to work on in the future.
As it stands now, everything works as intended, which is why I create this PR now, despite the known issues. Nevertheless, I understand if we only merge once at least issue 2 is resolved.
Please checkout the branch and really test the pipeline when reviewing. I worked on my own for a long time, so some things might now work on a different device, that I might not have noticed.
This closes #53.

# Note
Before starting the server, the database needs to be updated, this means executing
`python manage.py makemigrations` and `python manage.py migrate`.

### Issues:
1. Because some functions (e.g. the test mode for the UI) still rely on csv/xes files, these files are still written to disk. In a future update, all these functions need to be adapted to use database queries instead of reading files.
2. When users decide to discard the extraction results, the `Cohort` will still remain in the database, because it is always saved once extracted. This will affect all results, that have a relationship to the extracted `Trace`, right now `Cohort` is the only one implemented.
3. The names of patient journeys are unique, but it is not yet possible for users to choose a previously uploaded patient journey to extract from.

### Possible Solutions:
1. Add the necessary entries to the database and push it. From then on, the database file should not be included in commits.
2. Change the `execute_and_save()` method to return a dictionary instead. Then the `Orchestrator` can handle saving the model instances all together – if the user wishes to do so. (Django can create model instances directly from dictionaries.). 
3. Create a new view, complementary to the `JourneyInputView` in which all available patient journeys in the database are displayed.

### CODE REVIEW

**Comment by nils-schmitt** [comment]
It would be good to either automate the update of the db or include the update steps in the README

File: tracex/extraction/migrations/0005_alter_trace_managers_remove_trace_duration_and_more.py
Comment (nils-schmitt) [inline tracex/extraction/migrations/0005_alter_trace_managers_remove_trace_duration_and_more.py:1]: "and_more" seems like an odd name?

**Review by nils-schmitt — CHANGES_REQUESTED** [review]
Looks good so far, I would like to add onto the known issues, that only activities with retrieved eventtypes and location are shown in the result view. (But that should be tackled as an individual issue)

File: tracex/extraction/migrations/0005_alter_trace_managers_remove_trace_duration_and_more.py
Comment (FR-SON) [reply tracex/extraction/migrations/0005_alter_trace_managers_remove_trace_duration_and_more.py:1]: These files are automatically generated when running `manage.py makemigrations`, though i agree - it's an odd name :D

Comment (FR-SON) [reply tracex/extraction/migrations/0005_alter_trace_managers_remove_trace_duration_and_more.py:1]: Good point, I will update the README

**Comment by FR-SON** [comment]
> It would be good to either automate the update of the db or include the update steps in the README.
> 
> EDIT: You should also note, that if no executionpath is set, the command needs to be `python tracex/manage.py [...]` to run it from the root directory.

Good point, I will update the README

File: tracex/extraction/migrations/0005_alter_trace_managers_remove_trace_duration_and_more.py
Comment (nils-schmitt) [reply tracex/extraction/migrations/0005_alter_trace_managers_remove_trace_duration_and_more.py:1]: Okay, nevermind then

**Review by nils-schmitt — APPROVED** [review]

**Review by tkv29 — CHANGES_REQUESTED** [review]

**Review by tkv29 — APPROVED** [review]

**Pull request merged**